js_modules/dagit/packages/core/src/partitions/__tests__/CreatePartitionDialog.test.tsx

```diff
@@ -0,0 +1,62 @@
+import {MockedProvider, MockedResponse} from '@apollo/client/testing';
+import {act, render, screen, waitFor} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import * as React from 'react';
+
+import {CreatePartitionDialog} from '../CreatePartitionDialog';
+import {buildCreatePartitionFixture} from '../__fixtures__/CreatePartitionDialog.fixture';
+
+let onCloseMock = jest.fn();
+let onSelectedMock = jest.fn();
+
+function Test({mocks}: {mocks?: MockedResponse[]}) {
+  return (
+    <MockedProvider mocks={mocks}>
+      <CreatePartitionDialog
+        isOpen={true}
+        close={onCloseMock}
+        repoAddress={{
+          name: 'testing',
+          location: 'testing',
+        }}
+        partitionDefinitionName="testPartitionDef"
+        selected={['test1']}
+        setSelected={onSelectedMock}
+      />
+    </MockedProvider>
+  );
+}
+
+describe('CreatePartitionDialog', () => {
+  beforeEach(() => {
+    onCloseMock = jest.fn();
+    onSelectedMock = jest.fn();
+  });
+
+  it('Submits mutation with correct variables and calls setSelected and onClose', async () => {
+    const createFixture = buildCreatePartitionFixture({
+      partitionsDefName: 'testPartitionDef',
+      partitionKey: 'testPartitionName',
+    });
+    await act(async () => {
+      render(<Test mocks={[createFixture]} />);
+    });
+    const partitionInput = screen.getByTestId('partition-input');
+    userEvent.type(partitionInput, 'testPartitionName');
+    userEvent.click(screen.getByTestId('save-partition-button'));
+    await waitFor(() => {
+      expect(onCloseMock).toHaveBeenCalled();
+      expect(onSelectedMock).toHaveBeenCalledWith(['test1', 'testPartitionName']);
+      expect(createFixture.result).toHaveBeenCalled();
+    });
+  });
+
+  it('Shows error state', async () => {
+    await act(async () => {
+      render(<Test mocks={[]} />);
+    });
+    const partitionInput = screen.getByTestId('partition-input');
+    userEvent.type(partitionInput, 'invalidname\n\r\t');
+    expect(screen.getByTestId('warning-icon')).toBeVisible();
```

Comment:
Small nit but it'd be nice to verify that the warning icon isn't visible before the typing, and have a second bit here that puts in a valid name and then verifies it's no longer visible  (since there's a useMemo involved in the state change)